src/System.Net.Sockets/src/System/Net/Sockets/UDPClient.netstandard17.cs

```diff
@@ -0,0 +1,245 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace System.Net.Sockets
+{
+    public partial class UdpClient
+    {
+        public UdpClient(string hostname, int port)
+        {
+            if (hostname == null)
+            {
+                throw new ArgumentNullException(nameof(hostname));
+            }
+            if (!TcpValidationHelpers.ValidatePortNumber(port))
+            {
+                throw new ArgumentOutOfRangeException(nameof(port));
+            }
+            //
+            // NOTE: Need to create different kinds of sockets based on the addresses
+            //       returned from DNS. As a result, we defer the creation of the 
+            //       socket until the Connect method.
+            //
+            //CreateClientSocket();
+            Connect(hostname, port);
+        }
+
+        public void Close()
+        {
+            Dispose(true);
+        }
+
+        public void Connect(string hostname, int port)
+        {
+            if (_cleanedUp)
+            {
+                throw new ObjectDisposedException(this.GetType().FullName);
```

Comment:
In this case, I like the explicit `this`.  For some reason a naked `GetType` call confuses me. :)